gateway/gateway.go

```diff
@@ -27,14 +31,74 @@ type gateway struct {
 	mux               *http.ServeMux
 }
 
+// CollectionOperationsInterceptor parse collection operators and stores in corresponding message fields
+func CollectionOperationsInterceptor(parentCtx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
```

Comment:
It would better to have postfix `ClientUnaryInterceptor`, or probably move it to the `query` package at all .

Reply:
CollectionOperationsInterceptor depends on non public variables like fieldsQueryKey